File: src/network/packet/impl/response/ServerHandshakeResponsePacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\response;

use pocketcloud\cloud\network\packet\data\VerifyStatus;
use pocketcloud\cloud\network\packet\ResponsePacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class ServerHandshakeResponsePacket extends ResponsePacket {

    public function __construct(private readonly ?VerifyStatus $verifyStatus = null) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->verifyStatus);
    }

    public static function create(VerifyStatus $verifyStatus): self {
        return new self($verifyStatus);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/response/ServerHandshakeResponsePacket.java
package de.pocketcloud.cloud.network.packet.impl.response;

import de.pocketcloud.cloud.network.packet.ResponsePacket;
import de.pocketcloud.network.packet.type.VerificationStatus;
import de.pocketcloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class ServerHandshakeResponsePacket extends ResponsePacket {

    private VerificationStatus verificationStatus;

    public ServerHandshakeResponsePacket(VerificationStatus verificationStatus) {
        this.verificationStatus = verificationStatus;
    }

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(verificationStatus);
    }

    public static ServerHandshakeResponsePacket create(VerificationStatus verificationStatus) {
        return new ServerHandshakeResponsePacket(verificationStatus);
    }
}
